### PR TITLE
[es-shims] Add `Reflect.apply` polyfill

### PR DESCRIPTION
**Pull request by nicolo-ribaudo**

### CODE REVIEW

File: packages/babel-plugin-polyfill-es-shims/test/fixtures/usage-pure/static-methods/stderr.txt [hunk 1]
```diff
@@ -1,4 +1,4 @@
...
+	npm install --save array.from@^1.1.0 array.of@^1.0.0 is-nan@^1.2.1 math.acosh@^1.0.0 math.atanh@^1.0.0 math.cbrt@^1.0.0 math.clz32@^1.0.0 math.fround@^1.0.0 math.log1p@^1.0.1 math.sign@^2.0.0 object-is@^1.1.2 object.assign@^4.1.0 object.entries@^1.1.1 object.fromentries@^2.0.2 object.getownpropertydescriptors@^2.1.0 object.getprototypeof@^1.0.1 object.values@^1.1.1 reflect.apply@^1.0.0 reflect.getprototypeof@^1.0.0 reflect.ownkeys@^1.0.1 string.fromcodepoint@^1.0.0 string.raw@^1.0.1
+	yarn add array.from@^1.1.0 array.of@^1.0.0 is-nan@^1.2.1 math.acosh@^1.0.0 math.atanh@^1.0.0 math.cbrt@^1.0.0 math.clz32@^1.0.0 math.fround@^1.0.0 math.log1p@^1.0.1 math.sign@^2.0.0 object-is@^1.1.2 object.assign@^4.1.0 object.entries@^1.1.1 object.fromentries@^2.0.2 object.getownpropertydescriptors@^2.1.0 object.getprototypeof@^1.0.1 object.values@^1.1.1 reflect.apply@^1.0.0 reflect.getprototypeof@^1.0.0 reflect.ownkeys@^1.0.1 string.fromcodepoint@^1.0.0 string.raw@^1.0.1
```
Comment (JLHwung) [inline packages/babel-plugin-polyfill-es-shims/test/fixtures/usage-pure/static-methods/stderr.txt:4]: Unrelated to this PR: we should come up with a webpage rendering these commands. Copying & pasting multiline text on the terminal is not good.

Comment (nicolo-ribaudo) [reply packages/babel-plugin-polyfill-es-shims/test/fixtures/usage-pure/static-methods/stderr.txt:4]: The problem is that this command depends on your code, we cannot pre-write it on the website.

Usually with an update or code change you'll only need to install a small number of new polyfills.

Comment (JLHwung) [reply packages/babel-plugin-polyfill-es-shims/test/fixtures/usage-pure/static-methods/stderr.txt:4]: Oh I meant to some dynamic reports like
```
https://babel.dev/error?lz_string=(lz of missing depedencies, error kind)
```

Like AngularJS did in https://code.angularjs.org/1.8.0/docs/error/$compile/ctreq?p0=dash&p1=someDir

Comment (nicolo-ribaudo) [reply packages/babel-plugin-polyfill-es-shims/test/fixtures/usage-pure/static-methods/stderr.txt:4]: Oh so we can compress the list of deps and modern terminals allow clicking on URLs :+1:

**Pull request merged**